src/cpu.rs

```diff
@@ -236,6 +331,224 @@ impl CPU {
+    fn test_execute_br() {
+        let mut cpu = CPU::new();
+        cpu.cond = ConditionFlags::ZRO.into();
+        let opcode = Opcode::OP_BR {
+            n: false,
+            z: true,
+            p: false,
+            offset: 1,
+        };
+        cpu.execute(opcode).unwrap();
+        assert_eq!(cpu.pc, 0x3001);
+    }
+
+    #[test]
+    fn test_execute_jmp() {
+        let mut cpu = CPU::new();
+        cpu.update_register(1, 0x1234).unwrap();
+        let opcode = Opcode::OP_JMP { base_r: 1 };
+        cpu.execute(opcode).unwrap();
+        assert_eq!(cpu.pc, 0x1234);
+    }
+
+    #[test]
+    fn test_execute_jsr() {
+        let mut cpu = CPU::new();
+        let opcode = Opcode::OP_JSR { offset: 1 };
+        cpu.execute(opcode).unwrap();
+        assert_eq!(cpu.pc, 0x3001);
+        assert_eq!(cpu.r7, 0x3000);
+    }
+
+    #[test]
+    fn test_execute_jsrr() {
+        let mut cpu = CPU::new();
+        cpu.update_register(1, 0x1234).unwrap();
+        let opcode = Opcode::OP_JSRR { base_r: 1 };
+        cpu.execute(opcode).unwrap();
+        assert_eq!(cpu.pc, 0x1234);
+        assert_eq!(cpu.r7, 0x3000);
+    }
+
+    #[test]
+    fn test_execute_ld() {
+        let mut cpu = CPU::new();
+        cpu.memory.write(0x3001, 0x1234).unwrap();
+        let opcode = Opcode::OP_LD { dr: 0, offset: 1 };
+        cpu.execute(opcode).unwrap();
+        assert_eq!(cpu.r0, 0x1234);
+    }
+
+    #[test]
+    fn test_execute_ldi() {
+        let mut cpu = CPU::new();
+        cpu.memory.write(0x3001, 0x3002).unwrap();
+        cpu.memory.write(0x3002, 0x1234).unwrap();
+        let opcode = Opcode::OP_LDI { dr: 0, offset: 1 };
+        cpu.execute(opcode).unwrap();
+        assert_eq!(cpu.r0, 0x1234);
+    }
+
+    #[test]
+    fn test_execute_ldr() {
+        let mut cpu = CPU::new();
+        cpu.update_register(1, 0x3001).unwrap();
+        cpu.memory.write(0x3002, 0x1234).unwrap();
+        let opcode = Opcode::OP_LDR {
+            dr: 0,
+            base_r: 1,
+            offset: 1,
+        };
+        cpu.execute(opcode).unwrap();
+        assert_eq!(cpu.r0, 0x1234);
+    }
+
+    #[test]
+    fn test_execute_lea() {
+        let mut cpu = CPU::new();
+        let opcode = Opcode::OP_LEA { dr: 0, offset: 1 };
+        cpu.execute(opcode).unwrap();
+        assert_eq!(cpu.r0, 0x3001);
+    }
+
+    #[test]
+    fn test_execute_st() {
+        let mut cpu = CPU::new();
+        cpu.update_register(0, 0x1234).unwrap();
+        let opcode = Opcode::OP_ST { sr: 0, offset: 1 };
+        cpu.execute(opcode).unwrap();
+        assert_eq!(cpu.memory.read(0x3001).unwrap(), 0x1234);
+    }
+
+    #[test]
+    fn test_execute_sti() {
+        let mut cpu = CPU::new();
+        cpu.memory.write(0x3001, 0x3002).unwrap();
+        cpu.update_register(0, 0x1234).unwrap();
+        let opcode = Opcode::OP_STI { sr: 0, offset: 1 };
+        cpu.execute(opcode).unwrap();
+        assert_eq!(cpu.memory.read(0x3002).unwrap(), 0x1234);
```

Comment:
Cool tests! Maybe I would add tests that tested instructions working together.

Although these are also  always useful!